tools/test.py

```diff
@@ -1777,7 +1778,14 @@ def should_keep(case):
       t = FormatTimedelta(entry.duration)
       sys.stderr.write("%4i (%s) %s\n" % (i, t, entry.GetLabel()))
 
-  return result
+  if result['allPassed']:
+    print('All tests passed.')
+  else:
+    print('Failed tests:')
+    for failedTest in result['failed']:
+      print(EscapeCommand(failedTest.command))
+
+  return exitCode
```

Comment:
```suggestion
    print(f'{len(failed_tests)} failed tests:')
    print('\n'.join(EscapeCommand(failed_test.command) for failed_test in failed_tests))

  return exit_code
```

Reply:
The previous line says how many failed tests there were.